core/src/main/java/com/gentics/mesh/plugin/manager/MeshPluginManagerImpl.java

```diff
@@ -149,20 +226,69 @@ public void startPlugins() {
 					plugin.start();
 					// Set state for PF4J
 					pluginWrapper.setPluginState(PluginState.STARTED);
-					if (plugin instanceof MeshPlugin) {
-						// Set status for Mesh
-						MeshPlugin meshPlugin = (MeshPlugin) plugin;
-						setStatus(meshPlugin.id(), PluginStatus.STARTED);
-						pluginRegistry.preRegister(meshPlugin);
-					}
-					startedPlugins.add(pluginWrapper);
 
-					firePluginStateEvent(new PluginStateEvent(this, pluginWrapper, pluginState));
+					registerMeshPlugin(pluginWrapper.getPluginId());
 				} catch (Throwable e) {
 					log.error("Error while starting plugins " + e.getMessage(), e);
 				}
 			}
 		}
+		if (pluginStorage == restartingPlugins) {
+			pluginStorage.clear();
+		}
+	}
+
+	/**
+	 * Start the specified plugin and its dependencies.
+	 */
+	@Override
+	public PluginState startPlugin(String pluginId) {
```

Comment:
This method is copied from `super` impl, with the modification of the *Mesh* plugin status notification - it should be performed only after the plugin is initialized, as we now have the separate state for initialization..